README.md

```diff
@@ -119,10 +108,61 @@ or the full way using a struct with all the possible options for the page call
   |> Segment.Analytics.page
 ```
 
+
+### Using the Segment Context
+
+If you want to set the Context manually you should create a `Segment.Analytics.Context` struct with `Segment.Analytics.Context.new/1` 
+
+```
+context = Segment.Analytics.Context.new(%{active: false})
+
+Segment.Analytics.track(user_id, event, %{property1: "", property2: ""}, context)
+```
+
+## Configuration
+
+The library has a number of configuration options you can use to overwrite default values and behaviours
+
+* `config :segment, :sender_impl` Allows selection of a sender implementation. At the moment this defaults to `Segment.Analytics.Batcher` which will send all events in batch. Change this value to `Segment.Analytics.Sender` to have all messages sent immediately (asyncronously)
+* `config :segment, :max_batch_size` The maximum batch size of messages that will be sent to Segment at one time. Default value is 100.
+* `config :segment, :batch_every_ms` The time (in ms) between every batch request. Default value is 2000 (2 seconds)
+* `config :segment, :retry_attempts` The number of times to retry sending against the segment API. Default value is 3 
+* `config :segment, :retry_expiry` The maximum time (in ms) spent retrying. Default value is 10000 (10 seconds)
+* `config :segment, :retry_start` The time (in ms) to start the first retry. Default value is 100
+* `config :segment, :send_to_http` If set to `false`, the libray will override the Tesla Adapter implementation to only log segment calls to `debug` but not make any actual API calls. This can be useful if you want to switch off Segment for test or dev. Default value is true
+* `config :segment, :tesla, :adapter` This config option allows for overriding the HTTP Adapter for Tesla (which the library defaults to Hackney).This can be useful if you prefer something else, or want to mock the adapter for testing.
+
+## Usage in Phoenix
+
+This is how I add to a Phoenix project (may not be your preferred way)
+
+1. Add the following to deps section of your mix.exs: `{:segment, "~> 0.2.0"}`
+   and then `mix deps.get`
+2. Add segment to applications list in the Phoenix project mix.exs
+ie.
+```
```

Comment:
All these longer code blocks should do be three backticks plus the word elixir like '```elixir' to tell the syntax highlighter to correctly highlight.